app/chat_search/ai_services/service.py

```diff
@@ -0,0 +1,134 @@
+import re
+from typing import Any
+from .initialized_huggingface_embeddings import InitializedHuggingFaceEmbeddings
+from langchain.embeddings.huggingface import HuggingFaceEmbeddings
+from langchain.document_loaders.csv_loader import CSVLoader
+from langchain.retrievers import BM25Retriever, EnsembleRetriever
+from langchain.vectorstores import FAISS
+from langchain.chains import RetrievalQA
+from pydantic import BaseModel
+
+from app.aimodels.gpt4all.ai_services.completion_inference import (
+    CompletionInference,
+    CompletionInferenceInputs,
+)
+from app.chat_search.ai_services.marco_rerank_retriever import MarcoRerankRetriever
+from app.core.errors import ValidationError
+from app.core.minio import download_pickled_object_from_minio
+from app.core.model_cache import MODEL_CACHE_BASEDIR
+
+from sqlalchemy.orm import Session
+from minio import Minio
+from app.aimodels.bertopic.crud import bertopic_embedding_pretrained
+
+from sample_data import CHAT_DATASET_1_PATH
+
+
+class RetrievalService(BaseModel):
+    completion_inference: CompletionInference
+    db: Session | None = None
+    s3: Minio | None = None
+    sentence_model: Any | None = None  #: :meta private:
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    def retrieve(self, api_inputs: CompletionInferenceInputs, summarize=False):
+        # validate input
+        if not isinstance(api_inputs, CompletionInferenceInputs):
+            raise ValidationError("must input type CompletionInferenceInputs")
+
+        retriever = self._build_retriever(channel_names=[CHAT_DATASET_1_PATH])
+
+        if summarize:
+            llm = self.completion_inference._build_llm(api_inputs)
+            results = self._retrieve_and_summarize(
+                llm,
+                query=api_inputs.prompt,
+                retriever=retriever,
+            )
+        else:
+            results = self._retrieve_only(
+                query=api_inputs.prompt,
+                retriever=retriever,
+            )
+
+        return results
+
+    def _build_retriever(
+        self,
+        channel_names=[],
+    ):
+        # TODO: pull from minio
+        # model_name = os.path.join(MODEL_CACHE_BASEDIR, "all-MiniLM-L6-v2")
+        if self.sentence_model is None:
+            try:
+                sentence_model_db_obj = bertopic_embedding_pretrained.get_by_sha256(
+                    db=self.db,
+                    sha256="ad2efe50dfaeea5243da9476d25249496872aa3dd8bfa5a3bbd05014f2822abc"
```

Comment:
@erob123 TODO use embeddingpretrainedmodel/get by model name instead of sha https://github.com/orgs/MIT-AI-Accelerator/projects/2/views/1?pane=issue&itemId=35692135